server/src/main/java/com/paypal/selion/grid/servlets/transfer/ManagedArtifact.java

```diff
@@ -20,7 +20,7 @@
  * call.
  * 
  */
-public interface ManagedArtifact {
+public interface ManagedArtifact<U extends Criteria> {
```

Comment:
@shelar With the other changes you have made, I think we should be able to avoid turning this into a template type